src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -197,6 +172,90 @@ public sealed override void Initialize(AnalysisContext analysisContext)
                         context.ReportDiagnostic(type.CreateDiagnostic(Rule, type.FormatMemberName()));
                     }
                 });
+
+                return;
+
+                // Local functions
+
+                bool CanBeCoClassAttributeContext(INamedTypeSymbol type)
+                     => coClassAttributeSymbol != null && type.TypeKind == TypeKind.Interface;
+
+                INamedTypeSymbol? FindTypeIfCoClassAttribute(AttributeData attribute)
+                {
+                    RoslynDebug.Assert(coClassAttributeSymbol != null);
+
+                    if (attribute.AttributeClass.Equals(coClassAttributeSymbol) &&
+                        attribute.ConstructorArguments.Length == 1 &&
+                        attribute.ConstructorArguments[0].Kind == TypedConstantKind.Type &&
+                        attribute.ConstructorArguments[0].Value is INamedTypeSymbol typeSymbol &&
+                        typeSymbol.TypeKind == TypeKind.Class)
+                    {
+                        return typeSymbol;
+                    }
+
+                    return null;
+                }
+
+                bool CanBeDesignerAttributeContext(INamedTypeSymbol type)
+                    => designerAttributeSymbol != null && (type.TypeKind == TypeKind.Interface || type.TypeKind == TypeKind.Class);
+
+                INamedTypeSymbol? FindTypeIfDesignerAttribute(AttributeData attribute, Compilation compilation)
+                {
+                    RoslynDebug.Assert(designerAttributeSymbol != null);
+
+                    if ((attribute.ConstructorArguments.Length != 1 && attribute.ConstructorArguments.Length != 2) ||
+                        !attribute.AttributeClass.Equals(designerAttributeSymbol))
+                    {
+                        return null;
+                    }
+
+                    switch (attribute.ConstructorArguments[0].Value)
+                    {
+                        case string designerTypeName when designerTypeName != null:
```

Comment:
The when is redundant here.

```suggestion
                        case string designerTypeName:
```